Quaver.Shared/Screens/Select/SelectScreen.cs

```diff
@@ -536,5 +556,139 @@ public void SelectRandomMap()
+
+            // Switch statement for setting predefined variable to the correct file path depending on the select container type.
+            switch (type)
+            {
+                case SelectContainerStatus.Mapsets:
+                    filePath = Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory);
+                    break;
+
+                case SelectContainerStatus.Difficulty:
+                    var mapsetPath = Path.Combine(ConfigManager.SongDirectory.Value, selectedMapset.Directory);
+                    var difficultyPath = Path.Combine(mapsetPath, selectedMapset.Maps[selectedDifficultyIndex].Path);
+
+                    filePath = selectedMapset.Maps.Count > 1 ? difficultyPath : mapsetPath;
+                    break;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            // Commence deleting and reloading.
+            var confirmDelete = new ConfirmCancelDialog($"Are you sure you want to delete this {( type == SelectContainerStatus.Mapsets ? "mapset" : "difficulty" )} ? ", (sender, confirm) =>
+            {
+                var mapTitle = selectedMapset.Maps.First().Title;
+
+                // Externally loaded map check
+                if (MapManager.Selected.Value.Game != MapGame.Quaver)
+                {
+                    // Display error message
+                    NotificationManager.Show(NotificationLevel.Error, "This map was loaded from an external source, it cannot be deleted.");
+                    return;
+                }
+
+                // Dispose of the background for the currently selected map.
+                BackgroundHelper.Background?.Dispose();
+
+                // Dispose of the currently playing track.
+                AudioEngine.Track?.Dispose();
+
+                // Run path deletion in the background.
+                ThreadScheduler.Run(() => DeletePath(filePath));
+
+                // Remove mapset/difficulty from cache and AvailableMapsets list.
+                switch (type)
+                {
+                    case SelectContainerStatus.Mapsets:
+                        selectedMapset.Maps.ForEach(MapDatabaseCache.RemoveMap);
+                        MapManager.Mapsets.RemoveAt(selectedMapsetIndex);
+                        AvailableMapsets.Remove(selectedMapset);
+                        view.MapsetScrollContainer.InitializeWithNewSets();
+
+                        // Restore index
+                        if (selectedMapsetIndex == AvailableMapsets.Count)
+                            view.MapsetScrollContainer.SelectMapset(AvailableMapsets.Count - 1); // set to end
+                        else
+                            view.MapsetScrollContainer.SelectMapset(selectedMapsetIndex);
+                        break;
+
+                    case SelectContainerStatus.Difficulty:
+                        if (selectedMapset.Maps.Count > 1)
```

Comment:
Instead of checking this condition in already two places and duplicating all the code from the previous case branch, check it once in the beginning of the function and condition on that instead of `type` + checking in the Difficulty case branch.